src/decide.ts

```diff
@@ -1,17 +1,83 @@
 import { PostHog } from './posthog-core'
-import { Compression, DecideResponse } from './types'
+import { Compression, DecideResponse, RemoteConfig } from './types'
 import { STORED_GROUP_PROPERTIES_KEY, STORED_PERSON_PROPERTIES_KEY } from './constants'
 
 import { logger } from './utils/logger'
-import { document } from './utils/globals'
+import { assignableWindow, document } from './utils/globals'
 
 export class Decide {
     constructor(private readonly instance: PostHog) {
         // don't need to wait for `decide` to return if flags were provided on initialisation
         this.instance.decideEndpointWasHit = this.instance._hasBootstrappedFeatureFlags()
     }
 
+    private _loadRemoteConfigJs(cb: (config?: RemoteConfig) => void): void {
+        if (assignableWindow.__PosthogExtensions__?.loadExternalDependency) {
+            assignableWindow.__PosthogExtensions__?.loadExternalDependency?.(this.instance, 'remote-config', () => {
+                return cb(assignableWindow._POSTHOG_CONFIG)
+            })
+        } else {
+            logger.error('PostHog Extensions not found. Cannot load remote config.')
+            cb()
+        }
+    }
+
+    private _loadRemoteConfigJSON(cb: (config?: RemoteConfig) => void): void {
+        this.instance._send_request({
+            method: 'GET',
+            url: this.instance.requestRouter.endpointFor('assets', `/array/${this.instance.config.token}/config`),
+            callback: (response) => {
+                cb(response.json as RemoteConfig | undefined)
+            },
+        })
+    }
+
     call(): void {
+        // Call decide to get what features are enabled and other settings.
+        // As a reminder, if the /decide endpoint is disabled, feature flags, toolbar, session recording, autocapture,
+        // and compression will not be available.
+        const disableRemoteCalls = !!this.instance.config.advanced_disable_decide
+
+        if (!disableRemoteCalls) {
+            // TRICKY: Reset any decide reloads queued during config.loaded because they'll be
+            // covered by the decide call right above.
+            this.instance.featureFlags.resetRequestQueue()
+        }
+
+        if (this.instance.config.__preview_remote_config) {
+            // Attempt 1 - use the pre-loaded config if it came as part of the token-specific array.js
+            if (assignableWindow._POSTHOG_CONFIG) {
+                logger.info('Using preloaded remote config', assignableWindow._POSTHOG_CONFIG)
+                this.onRemoteConfig(assignableWindow._POSTHOG_CONFIG)
+                return
+            }
+
+            if (disableRemoteCalls) {
+                logger.warn('Remote config is disabled. Falling back to local config.')
+                return
+            }
+
+            // Attempt 2 - if we have the external deps loader then lets load the script version of the config that includes site apps
+            this._loadRemoteConfigJs((config) => {
```

Comment:
this is tricky to test the unhappy path locally

if I turn on __preview_remote_config against current master then my local posthog returns a 200 with all the react boostrap which would show 404

but instead fails here because the mime type is unexpected

and then we call onRemoteConfig where I'd expect us to fallback to decide

other than that (admittedly edgy case) this all seems solid

Reply:
Yeah the changes for the config aren't in yet so it makes sense that that path fails. Feels okay given the whole point of this is for me to gradually test and improve it